src/galax/potential/_potential/builtin/builtin.py

```diff
@@ -394,6 +395,140 @@ def _potential(
+    ``positive_density=False``, this density will be negative in some regions,
+    but is a better approximation out to about 7 or 8 disk scale lengths.
+    """
+
+    m_tot: AbstractParameter = ParameterField(dimensions="mass")  # type: ignore[assignment]
+    """Total mass of the potential."""
+
+    h_R: AbstractParameter = ParameterField(dimensions="length")  # type: ignore[assignment]
+    """Radial (exponential) scale length."""
+
+    h_z: AbstractParameter = ParameterField(dimensions="length")  # type: ignore[assignment]
+    """
+    If ``sech2_z=True``, this is the scale height in a sech^2 vertical profile. If
+    ``sech2_z=False``, this is an exponential scale height.
+    """
+
+    positive_density: bool = eqx.field(default=False, static=True)
+    """
+    If ``True``, the density will be positive everywhere, but is only a good
+    approximation of the exponential density within about 5 disk scale lengths. If
+    ``False``, the density will be negative in some regions, but is a better
+    approximation out to about 7 or 8 disk scale lengths.
+    """
+
+    sech2_z: bool = eqx.field(default=False, static=True)
+    """
+    If ``True``, the vertical profile will approximate a sech^2 profile. If ``False``,
+    the vertical profile will approximate an exponential.
+    """
+
+    _: KW_ONLY
+    units: AbstractUnitSystem = eqx.field(converter=unitsystem, static=True)
+    constants: ImmutableMap[str, Quantity] = eqx.field(
+        default=default_constants, converter=ImmutableMap
+    )
+
+    def _get_mn_components(
+        self, t: gt.BatchableRealQScalar, /
+    ) -> list[MiyamotoNagaiPotential]:
+        hR = self.h_R(t)
+        hzR = (self.h_z(t) / hR).decompose(dimensionless).value
+        K = _mn3_K_pos_dens if self.positive_density else _mn3_K_neg_dens
+
+        # get b / h_R with fitting functions:
+        if self.sech2_z:
+            b_hR = -0.033 * hzR**3 + 0.262 * hzR**2 + 0.659 * hzR
+        else:
+            b_hR = -0.269 * hzR**3 + 1.08 * hzR**2 + 1.092 * hzR
+
+        x = jnp.vander(b_hR[None], N=5)[0]
+        param_vec = K @ x
+
+        # use fitting function to get the Miyamoto-Nagai component parameters
+        # NOTE(adrn): I had to add .value here otherwise when @jit'ing, this function
+        # would hang indefinitely...not sure why
+        mn_ms = param_vec[:3] * self.m_tot(t).value
+        mn_as = param_vec[3:] * hR.value
+        mn_b = b_hR * hR.value
+
+        return [
+            MiyamotoNagaiPotential(m_tot=m, a=a, b=mn_b, units=self.units)  # type: ignore[call-arg]
+            for m, a in zip(mn_ms, mn_as, strict=True)
+        ]
+
+    @partial(jax.jit)
+    def _potential(
+        self, q: gt.BatchQVec3, t: gt.BatchableRealQScalar, /
+    ) -> gt.SpecificEnergyBatchScalar:
+        return Quantity(
+            xp.sum(
+                xp.asarray(
+                    [mn.potential(q, t).value for mn in self._get_mn_components(t)]
```

Comment:
I think this might need to be 
```suggestion
                    [mn.potential(q, t).to_value(self.units["specific energy"]) for mn in self._get_mn_components(t)]
```
In which case store the out as an intermediate variable?